### DISCUSSION TITLE
Disable account linking I account console

### DISCUSSION THREAD

**Discussion by oculos** [keycloak/keycloak#28967] [Q&A] [answered]
Hi!

Is it possible to block users from linking accounts when we configure a realm with other providers? 

The thing is that we will use another realm as an IdP for admin users, so it would be nice to prevent guest users to attempt to link their accounts to those admin users.

Best,
Francis

**Comment by codespearhead** [comment]
Try going to `authentication > first broker login` in the left panel of the Keycloak Admin Console, then disable the "Confirm Link Existing Account" and "Handle Existing Account" actions.

**Reply by oculos** [reply]
Thanks! I’ll try that. But what I really want is to hide the providers from the user on the account console, so that the user won’t be tempted to attempt linking.

**Reply by hypesystem** [reply]
You can override the template files used for the account console to change what is generated. Could that be a solution?

**Reply by oculos** [reply]
> You can override the template files used for the account console to change what is generated. Could that be a solution?

As long as the user can't do it another way (API?), yes! :)

**Comment by danielFesenmeyer** [comment]
Afaik, with Keycloak default behavior ("Confirm Link Existing Account" etc), it's not possible to link an IDP without providing the Keycloak credentials of the user to be linked - which is the admin user in your case. Isn't that secure enough for your use case?

Not sure whether it's possible to provide a good user experience by disabling authenticators (meaningful error messages etc).

**Reply by oculos** [reply]
I know. My only goal here is to hide this for the user on his account console, not to block the possibility per se.

**Reply by danielFesenmeyer** [reply, marked as answer]
Found this in the docs, which could be helpful - there is a "Hide on Login Page" flag for each IDP: https://www.keycloak.org/docs/latest/server_admin/#_identity_broker:~:text=ON%20or%20OFF.-,Hide%20on%20Login%20Page,-When%20ON%2C%20Keycloak

**Reply by ssilvert** [reply]
> I know. My only goal here is to hide this for the user on his account console, not to block the possibility per se.

This tool we just released today might be helpful.  It allows you to create your own version of the account console.  Basically, you would just hack up the single page you want to change and let the tool build your customized console.

https://www.keycloak.org/2024/09/keycloak-customizable-ui

But if you only want to hide an entire page there is an even easier way.  Just extend the `keycloak.v2` theme and override the `content.json` file.

**Reply by cgeorgilakis** [reply]
> > I know. My only goal here is to hide this for the user on his account console, not to block the possibility per se.
> 
> This tool we just released today might be helpful. It allows you to create your own version of the account console. Basically, you would just hack up the single page you want to change and let the tool build your customized console.
> 
> https://www.keycloak.org/2024/09/keycloak-customizable-ui
> 
> But if you only want to hide an entire page there is an even easier way. Just extend the `keycloak.v2` theme and override the `content.json` file.

@ssilvert the problem is that although account has dedicated roles, Keycloak does not support granular permissions in Account Console. See [according ticket](https://github.com/keycloak/keycloak/issues/8734).
It would be great if someone could make a PR for this.
It is important having different privileges per environment and per user. Theme extension is not enough.
